src/System.Private.CoreLib/src/System/Runtime/CompilerServices/ClassConstructorRunner.cs

```diff
@@ -279,6 +279,13 @@ public static CctorHandle GetCctor(StaticClassConstructionContext* pContext)
 #else
                 const int Grow = 10;
 #endif
+
+                if (s_cctorGlobalLock == null)
```

Comment:
This should be under a WASM ifdef with TODO linking to an issue tracking the removal. This has a race condition and overall isn't great to keep around for long.

Reply:
Done. (Also fixed the race condition).